src/observerClass.ts

```diff
@@ -52,14 +63,7 @@ export function makeClassComponentObserver(
     }
     patch(target, "componentWillUnmount", function() {
         if (isUsingStaticRendering() === true) return
-        if (this.render[mobxAdminProperty]) {
-            this.render[mobxAdminProperty].dispose()
-        } else if (__DEV__) {
-            const displayName = getDisplayName(this)
-            console.warn(
-                `The render function for an observer component (${displayName}) was modified after MobX attached. This is not supported, since the new function can't be triggered by MobX.`
-            )
-        }
+        this.render[mobxAdminProperty] && this.render[mobxAdminProperty].dispose()
```

Comment:
```suggestion
        this.render[mobxAdminProperty]?.dispose()
```

Reply:
Yep, I'm planning on submitting a PR in the lite repo.